File: backend/app/domain/entities/session.py
"""セッションエンティティ"""

from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import Any


class SessionStatus(str, Enum):
    """セッションステータス"""

    ACTIVE = "active"
    INACTIVE = "inactive"
    ENDED = "ended"


@dataclass
class Session:
    """
    ユーザーセッションエンティティ

    チャットセッションを管理するドメインエンティティ
    """

    session_id: str
    user_id: str
    status: SessionStatus
    metadata: dict[str, Any] | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
    expires_at: int | None = None

    def __post_init__(self):
        """エンティティ作成後の初期化"""
        if not self.session_id:
            raise ValueError("session_idは必須です")
        if not self.user_id:
            raise ValueError("user_idは必須です")
        if self.status not in SessionStatus:
            raise ValueError(f"無効なstatus: {self.status}")

    def activate(self) -> None:
        """セッションをアクティブにする"""
        self.status = SessionStatus.ACTIVE
        self.updated_at = datetime.now()

    def deactivate(self) -> None:
        """セッションを非アクティブにする"""
        self.status = SessionStatus.INACTIVE
        self.updated_at = datetime.now()

    def end(self) -> None:
        """セッションを終了する"""
        self.status = SessionStatus.ENDED
        self.updated_at = datetime.now()

    def is_active(self) -> bool:
        """セッションがアクティブかどうか"""
        return self.status == SessionStatus.ACTIVE

    def update_metadata(self, metadata: dict[str, Any]) -> None:
        """メタデータを更新"""
        if self.metadata is None:
            self.metadata = {}
        self.metadata.update(metadata)
        self.updated_at = datetime.now()

File: flutter/lib/models/session.dart
/// チャットセッションモデル
class ChatSession {
  final String sessionId;
  final String userId;
  final String status;
  final DateTime? createdAt;
  final Map<String, dynamic>? metadata;

  const ChatSession({
    required this.sessionId,
    required this.userId,
    required this.status,
    this.createdAt,
    this.metadata,
  });

  factory ChatSession.fromJson(Map<String, dynamic> json) {
    return ChatSession(
      sessionId: json['session_id'] as String,
      userId: json['user_id'] as String? ?? '',
      status: json['status'] as String? ?? 'active',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      metadata: json['metadata'] as Map<String, dynamic>?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'session_id': sessionId,
      'user_id': userId,
      'status': status,
      'created_at': createdAt?.toIso8601String(),
      'metadata': metadata,
    };
  }

  bool get isActive => status == 'active';
}




